src/main/java/frc/robot/constants/RobotComponents.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.constants;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.RemoteSensorSource;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import frc.robot.components.Pigeon;
+import frc.robot.components.TrigonTalonSRX;
+import frc.robot.utilities.MotorConfig;
+import frc.robot.utilities.pid.PIDFTalonFX;
+
+public class RobotComponents {
+    protected static class SwerveComponents {
+        public static final Pigeon PIGEON = new Pigeon(RobotMap.CAN.SwerveMap.PIGEON_ID);
+
+        // configs
+        private static final MotorConfig ANGLE_MOTOR_CONFIG = new MotorConfig().
+                inverted(false).
+                sensorPhase(false).
+                withOpenLoopRampRate(2).
+                withClosedLoopRampRate(0.5).
+                brake().
+                withCurrentLimit(new SupplyCurrentLimitConfiguration(
+                        true,
+                        25,
+                        40,
+                        0.1
+                ));
+        private static final MotorConfig DRIVE_MOTOR_CONFIG = new MotorConfig().
+                inverted(true).
+                sensorPhase(false).
+                withOpenLoopRampRate(0.1).
+                withClosedLoopRampRate(0.1).
+                brake().
+                withCurrentLimit(new SupplyCurrentLimitConfiguration(
+                        true,
+                        34,
+                        60,
+                        0.1
+                ));
+        private static final MotorConfig ANGLE_ENCODER_CONFIG = new MotorConfig().
+                withFeedbackNotContinuous(true).
+                withFeedbackDevice(FeedbackDevice.CTRE_MagEncoder_Absolute);
+
+        /* Front Left */
+        public static final PIDFTalonFX FRONT_LEFT_ANGLE_MOTOR = new PIDFTalonFX(
```

Comment:
if the suggestions to shorten these constructors don't make it short enough make FRONT_LEFT_ANGLE_MOTOR_CONFIG. or some array. there is a better way to do this. or have the swerveConstants get the motor and pid and it adds it there.